### PR TITLE
Add showpenalty option

### PR DESCRIPTION
**Pull request by PaulWalther**
As we do not use penalties in our course we wanted to add this additional option to avoid showing the penalty to the students.

### CODE REVIEW

File: src/turnin.c [hunk 1]
```diff
@@ -168,7 +168,8 @@ void version() {
 	        "Copyright 2010-2014 Bryce Boe        <bboe@cs.ucsb.edu>\n"
 	        "Copyright 2014      Foivos S. Zakkak <foivos@zakkak.net> and\n"
 	        "                    Antonios Chariton<daknob.mac@gmail.com>\n"
-	        "Copyright 2015-2022 Foivos S. Zakkak <foivos@zakkak.net>\n\n"
+	        "Copyright 2015-2022 Foivos S. Zakkak <foivos@zakkak.net>\n"
+			"Copyright 2024 	 Paul Walther     <paul.walther@tum.de>\n\n"
```
Comment (zakkak) [inline src/turnin.c:172]:
```suggestion
	        "Copyright 2015-2022 Foivos S. Zakkak <foivos@zakkak.net>\n"
	        "Copyright 2024      Paul Walther     <paul.walther@tum.de>\n\n"
```

File: src/turnin.c [hunk 2]
```diff
@@ -634,7 +635,15 @@ void setup(char *arg) {
...
-			} else if (strcasecmp(keyword, "duedate") == 0) {
+			} else if (strcasecmp(keyword, "showpenalty") == 0) {
+				if ((n != 0) || (n != 1)) {
+					fprintf(stderr, "turnin: showpenalty must be either 0 or 1\n"
+					                "        Please notify the Instructor or a TA.\n");
+					exit(1);
+				}
+				showpenalty = n;
+			} 
+			else if (strcasecmp(keyword, "duedate") == 0) {
```
Comment (zakkak) [inline src/turnin.c:646]:
```suggestion
			} else if (strcasecmp(keyword, "duedate") == 0) {
```

File: src/turnin.c [hunk 3]
```diff
@@ -1385,7 +1394,9 @@ void checkdue() {
 		fprintf(stderr, "******************************************\n");
 		(void)fclose(fd);
 	}
-	fprintf(stderr, "\n*** This turn in will get %d%% penalty, due to late turn in, on the final grade ***\n", penalty);
+	if ((penalty > 0)&& (showpenalty)) {
```
Comment (zakkak) [inline src/turnin.c:1397]:
```suggestion
	if ((penalty > 0) && showpenalty) {
```

**Review by zakkak — CHANGES_REQUESTED** [review]
Hi @PaulWalther, thank you for your contribution.

It LGTM, could you just address the suggested styling issues before I can merge this? Thank you.

**Comment by PaulWalther** [comment]
Hey @zakkak . Sure, should have looked into the styleguide before opening the PR. I applied your changes. Best regards

**Pull request merged**